action.yml

```diff
@@ -8,35 +8,63 @@ inputs:
   integration_url:
     description: 'The OpsLevel deploy integration url'
     required: true
-    default: ''
-  service:
-    description: 'The service alias for the event'
+  deduplication_id:
+    description: 'An identifier that can be used to deduplicate deployments'
+    required: false
+    default: ${{ github.run_id }}
+  deployer_email:
+    description: 'The deployer email who created the event'
     required: false
     default: ''
+  deployer_name:
+    description: 'The deployer name who created the event'
+    required: false
+    default: ${{ github.actor }}
   description:
     description: 'The description or release notes for the event'
     required: false
-    default: ''
+    default: 'Event Created by OpsLevel Github Action'
   environment:
     description: 'The environment for the event'
     required: false
-    default: ''
+    default: production
   number:
-    description: 'The deploy number for the event - Default: ${GITHUB_RUN_NUMBER}'
+    description: 'The deploy number for the event'
     required: false
-    default: ''
-  deployer_name:
-    description: 'The deployer name who created the event - Default: ${GITHUB_ACTOR}'
-    required: false
-    default: ''
-  deployer_email:
-    description: 'The deployer email who created the event'
-    required: false
-    default: ''
-  deduplication_id:
-    description: 'An identifier that can be used to deduplicate deployments'
+    default: ${{ github.run_number }}
+  service:
+    description: 'The service alias for the event'
     required: false
-    default: ''
+    default: ${{ github.repository }}
 runs:
-  using: 'docker'
-  image: 'Dockerfile'
+  using: "composite"
+  steps:
+    - name: Report Deploy to OpsLevel
```

Comment:
Is there a reason you don't put a step right before this that installs the CLI?

Reply:
Heres a first pass chatGPT took at it.  It looks about 85-90% correct but we probably need to tweak it after testing.  We should probably put this in the CLI repo and then you don't have to use homebrew and then you can do the easy step here of
```
curl -sLS https://raw.githubusercontent.com/OpsLevel/cli/main/install.sh | sudo sh
```

Here is the script
```
#!/bin/bash

# Version of the OpsLevel CLI to install
VERSION=${1:-v2024.07.1} # Default version, override by passing as an argument

# Function to determine the OS
detect_os() {
    case "$(uname -s)" in
        Linux*)     OS=linux;;
        Darwin*)    OS=macos;;
        CYGWIN*|MINGW*|MSYS*|MINGW32*|MINGW64*) OS=windows;;
        *)          echo "Unsupported OS"; exit 1;;
    esac
    echo "Detected OS: $OS"
}

# Function to determine the architecture
detect_arch() {
    case "$(uname -m)" in
        x86_64)    ARCH=amd64;;
        armv7l)    ARCH=armv7;;
        aarch64)   ARCH=arm64;;
        *)         echo "Unsupported architecture"; exit 1;;
    esac
    echo "Detected Architecture: $ARCH"
}

# Function to download and install the CLI tool
install_cli() {
    DOWNLOAD_URL="https://github.com/OpsLevel/cli/releases/download/${VERSION}/opslevel_${OS}_${ARCH}.tar.gz"
    
    # Temporary directory to store the download
    TMP_DIR=$(mktemp -d)

    echo "Downloading OpsLevel CLI from $DOWNLOAD_URL ..."
    curl -L -o "$TMP_DIR/opslevel.tar.gz" "$DOWNLOAD_URL"
    
    if [ $? -ne 0 ]; then
        echo "Download failed. Please check the version and try again."
        exit 1
    fi

    echo "Extracting the OpsLevel CLI..."
    tar -xzf "$TMP_DIR/opslevel.tar.gz" -C "$TMP_DIR"

    echo "Installing the OpsLevel CLI to /usr/local/bin ..."
    sudo mv "$TMP_DIR/opslevel" /usr/local/bin/

    if [ $? -ne 0 ]; then
        echo "Installation failed."
        exit 1
    fi

    echo "Cleaning up..."
    rm -rf "$TMP_DIR"

    echo "OpsLevel CLI installed successfully!"
}

# Main script execution
detect_os
detect_arch
install_cli

```

Reply:
Good point, I forgot to consider that the CLI adds commit information in the report deploy request 🤔